src/Data/Binary/Serialise/CBOR/Tutorial.hs

```diff
@@ -16,33 +16,159 @@ module Data.Binary.Serialise.CBOR.Tutorial
   ) where
 
+import Data.Binary.Serialise.CBOR
+
 {- $introduction
 
-Lorem ipsum...
+@binary-serialise-cbor@ is a library for the serialisation of Haskell values.
+
+As in modern modern serialisation libraries, @binary-serialise-cbor@ offers
+instance derivation via GHC's 'GHC.Generic' mechanism,
+
+> import qualified Data.Binary.Serialise.CBOR as CBOR
+> import qualified Data.ByteString.Lazy as BSL
+>
+> data Animal = HoppingAnimal { animalName :: String, hoppingHeight :: Int }
+>             | WalkingAnimal { animalName :: String, walkingSpeed :: Int }
+>             deriving (Generic)
+>
+> instance CBOR.Serialise Animal
+>
+> fredTheFrog :: Animal
+> fredTheFrog = HoppingAnimal "Fred" 4
+>
+> main = BSL.writeFile "hi" (CBOR.serialise fredTheFrog)
+
+We can then later read Fred,
+
+> main = do
+>     fred <- CBOR.deserialise <$> BSL.readFile "hi"
+>     print fred
 
 -}
 
 {- $cborformat
 
-Lorem ipsum...
+@binary-serialise-cbor@ uses the Concise Binary Object Representation, CBOR
+(IETF RFC 7049), as its serialised representation. This encoding is efficient in
+both encoding\/decoding complexity as well as space, and is generally
+machine-independent.
+
+The CBOR data model resembles that of JSON, having arrays, key/value maps,
+integers, floating point numbers, and a variety of string types. In addition,
+CBOR allows items to be /tagged/ with a number which describes the type of data
+that follows. This can be used both to identify which data constructor of a type
+an encoding represents, as well as representing different versions of the same
+constructor.
+
+*** A note on interoperability
+
+@binary-serialise-cbor@ is intended primarily as a /serialisation/ library for
+Haskell values. That is, a means of stably storing Haskell values for later
+reading by @binary-serialise-cbor@. While it uses the CBOR encoding format, the
+library doesn't /not/ primarily aim to facilitate serialisation and
+deserialisation across different CBOR implementations.
+
+If you want to use @binary-serialise-cbor@ to serialise\/deserialise values
+for\/from another CBOR implementation (either in Haskell or another language),
+you should keep a few things in mind,
+
+1. The 'Serialise' instances for some "basic" Haskell types (e.g. 'Maybe',
+   'Data.ByteString.ByteString') don't carry a tag, in contrast to common
+   convention. This is an intentional design decision to minimize encoding size.
+
+2. The library reserves the right to change encodings in
+   non-backwards-compatible ways across super-major versions.
+
+3. While the library tries to use standard encodings in its instances wherever possible,
+   these instances aren't guaranteed to implement all valid variants of the
+   encodings in the specification. For instance, the 'UTCTime' instance only
+   implements a small subset of the encodings described by the Extended Date
+   RFC.
 
 -}
 
+
 {- $encoding
 
-Lorem ipsum...
+During encoding, abstract CBOR resentations are represented by the
```

Comment:
I'd restructure this so we first do the worked example with generic deriving and then do the worked example for manual instances and talk about versioning/migration. Then we can talk about interop with other CBOR impls last.